pep_sphinx_extensions/__init__.py

```diff
@@ -37,12 +38,37 @@ def _depart_maths():
     pass  # No-op callable for the type checker
 
 
-def _update_config_for_builder(app: Sphinx):
+def _check_external_builder(app: Sphinx, conf: Config) -> None:
+    if "internal_builder" not in app.tags:  # internal_builder exists if Sphinx is run by build.py
+        conf.html_context = {"external_builder": True}
+    else:
+        conf.html_context = {"external_builder": False}
+    app.tags.remove("internal_builder")
+
+    if conf.html_context["external_builder"]:
+        app.connect("build-finished", _post_build)  # Post-build tasks
+
+
+def _update_config_for_builder(app: Sphinx) -> None:
+    if app.config.html_context["external_builder"]:
+        app.builder.copysource = False  # Prevent unneeded source copying - we link direct to GitHub
```

Comment:
Should this be set to True in the configuration instead being behind a conditional in the extension? Even for internal builds, I don't see why we'd want to show the sources.

https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_copy_source

Reply:
I can't quite remember why I set this as false in the script instead of in the config -- but it should always be false, this method `_update_config_for_builder` does all the stuff `build.py` would normally do.

Will look to see if I can move to the config, if so that would be simpler.